x-pack/plugins/ml/public/ui_actions/open_vis_in_ml_action.tsx

```diff
@@ -49,19 +52,16 @@ export function createVisToADJobAction(
         return Promise.reject();
       }
     },
-    async isCompatible(context: { embeddable: Embeddable }) {
-      const embeddableType = context.embeddable.type;
-      if (embeddableType !== 'map') {
-        if (embeddableType !== 'lens' || !context.embeddable.getSavedVis()) {
-          return false;
-        }
-      }
+    async isCompatible({ embeddable }: EmbeddableApiContext) {
+      if (!isApiCompatible(embeddable) || !['map', 'lens'].includes(embeddable.type)) return false;
```

Comment:
nit: should we use `apiIsOfType(api, 'lens') || apiIsOfType(api, 'map')` here? IMO It's better to be consistent.